test/onyx/validation_test.clj

```diff
@@ -490,3 +491,88 @@
                                     (keys (s/explain {(os/restricted-ns :myplugin) s/Any
                                                       :myplugin/option s/Str})))))
            :myplugin))))
+
+(def blank-job {:workflow []
+                :catalog []
+                :lifecycles []
+                :windows []
+                :triggers []
+                :flow-conditions []
+                :task-scheduler :onyx.task-scheduler/balanced})
+
+(deftest task-bundle-composition
+  (testing "task bundles without schema"
+    (let [task-bundle {:task {:task-map {:onyx/name :in
+                                         :onyx/plugin :onyx.plugin.core-async/input
+                                         :onyx/type :input
+                                         :onyx/medium :core.async
+                                         :onyx/batch-size 10
+                                         :onyx/max-peers 1}}}]
+      (is (add-task blank-job task-bundle))
+      (is (thrown? Exception (add-task blank-job {:task {:task-map {:onyx/name :bad-job}}})))))
+  (testing "task bundles with task-map schema"
+    (let [task-bundle {:task {:task-map {:onyx/name :in
+                                         :onyx/plugin :onyx.plugin.core-async/input
+                                         :onyx/type :input
+                                         :onyx/medium :core.async
+                                         :onyx/batch-size 10
+                                         :onyx/max-peers 1}}
+                       :schema {:task-map {:custom/thing s/Str}}}]
+      (is (add-task blank-job (assoc-in task-bundle [:task :task-map :custom/thing] "Hello")))
+      (is (thrown? Exception (add-task blank-job task-bundle)))))
+  (testing "task bundles with lifecycle schema"
+    (let [task-bundle {:task {:task-map {:onyx/name :in
+                                         :onyx/plugin :onyx.plugin.core-async/input
+                                         :onyx/type :input
+                                         :onyx/medium :core.async
+                                         :onyx/batch-size 10
+                                         :onyx/max-peers 1}
+                              :lifecycles [{:lifecycle/calls ::woo
+                                            :lifecycle/task :in}]}
+                       :schema {:lifecycles [{:custom/thing s/Str}]}}]
+      (is (add-task blank-job (assoc-in task-bundle [:task :lifecycles 0 :custom/thing] "Hello")))
+      (is (thrown? Exception (add-task blank-job task-bundle)))))
+  (testing "task bundles with window schema"
```

Comment:
For some reason the user supplied schema is not taken into account for validation.